src/include/migraphx/shape_for_each.hpp

```diff
@@ -51,7 +51,7 @@ void shape_for_each(const migraphx::shape& s, F f)
                            assert(len > 0 and stride > 0);
                            return (i / stride) % len;
                        });
-        call(indices);
+        f(index_const_ref);
```

Comment:
This wouldn't result in correct results. Function `f` needs to be called while calculating new indices as it was done previously.

Reply:
It is a const reference. That is what the lambda was doing, while adding an indirection of call.. Thanks.